Add packages and configuration changes to allow Luna to start in QEMU

magmastonealex opened the pull request:
Luna will now start in QEMU. To do so (either on this branch, or once everything's in master):
- Install postmarketos-ui-luna
- Run /usr/bin/start_luna.sh
- The first-use app should appear. Complete the app (Click on where the "next" button should be for the step after license agreements that just has a black screen).
- Run `luna-send -n 1 'luna://com.palm.configurator/run' '{"types":["dbkinds","dbpermissions","activities"]}'` in another terminal
- Stop luna, and reboot.
- Run /usr/bin/start_luna.sh

Much of the race conditions that require the reboots, manual `configurator` running, and long delays in `start_luna.sh` can be dealt with using openRC startup files.

Most apps appear to work, with the notable exception of the email client (which is missing services that it needs).

ARM support is still broken because of #1008 and #878. 

(more work towards #629)

ollieparanoid commented on aports/luna/enyo/APKBUILD:
```diff
@@ -0,0 +1,26 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webOS-ports/enyo-1.0/archive/$_commit.tar.gz"
+builddir="$srcdir/$pkgname-1.0-$_commit"
+
+build() {
```
Empty `build()` functions can be removed.

ollieparanoid commented on aports/luna/luna-app-services/APKBUILD:
```diff
@@ -0,0 +1,89 @@
+	mkdir -p "$srcdir"/build
+}
+
+build() {
```
empty build function

ollieparanoid commented on aports/luna/luna-app-browser/APKBUILD:
```diff
@@ -0,0 +1,42 @@
+	install -d "$pkgdir"/etc/palm/db/permissions
+	install -v -m644 "$builddir"/configuration/db/kinds/* "$pkgdir"/etc/palm/db/kinds
+	install -v -m644 "$builddir"/configuration/db/permissions/* "$pkgdir"/etc/palm/db/permissions
+
```
unnecessary empty lines

ollieparanoid commented on aports/luna/luna-app-services/APKBUILD:
```diff
@@ -0,0 +1,89 @@
+	install -d "$pkgdir"/etc/palm/event.d
+	install -d "$pkgdir"/etc/palm/init
+
+    for SERVICE in `ls -d1 ./com.palm.service.*` ; do
```
That syntax is obsolete as far as I know, the new one would be:
```shell
for SERVICE in $(ls -d1 ./com.palm.service.*); do
```

ollieparanoid commented on aports/luna/luna-app-services/APKBUILD:
```diff
@@ -0,0 +1,89 @@
+	install -d "$pkgdir"/etc/palm/init
+
+    for SERVICE in `ls -d1 ./com.palm.service.*` ; do
+        SERVICE_DIR=`basename $SERVICE`
```
see above

montvid commented:
For testing purposes i installed Luna on a armhf Nexus 7 2013 (it has mesa support - hardware acceleration) here is the error log - reminds me of #878. I also needed to install x11 it seems to get no warnings/errors:

apk add xorg-server xf86-input-libinput libinput
https://paste.gnome.org/poiclprj6

ollieparanoid commented on aports/luna/luna-app-services/APKBUILD:
```diff
@@ -0,0 +1,89 @@
+
+	# create folder for contact linker plugins
+	mkdir -p "$pkgdir"/etc/palm/contact_linker_plugins
+
```
empty lines

ollieparanoid commented on aports/luna/luna-applauncher/APKBUILD:
```diff
@@ -0,0 +1,24 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webOS-ports/$pkgname/archive/$_commit.tar.gz"
+builddir="$srcdir/$pkgname-$_commit"
+
+build() {
```
empty build function

magmastonealex commented:
@montvid Correct, #878 means that you can't use Luna on armhf devices yet. You shouldn't need those x11 packages - there are some warnings without them, but everything should still work.

ollieparanoid commented on aports/luna/luna-core-apps/APKBUILD:
```diff
@@ -0,0 +1,53 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webOS-ports/core-apps/archive/$_commit.tar.gz"
+builddir="$srcdir/core-apps-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/luna-core-apps/APKBUILD:
```diff
@@ -0,0 +1,53 @@
+	
+	cd "$builddir"
+
+    install -d "$pkgdir"/usr/palm/applications
```
spaces instead of tabs

ollieparanoid commented on aports/luna/luna-core-apps/APKBUILD:
```diff
@@ -0,0 +1,53 @@
+        fi
+    done
+
+    # TODO: Upstream removes these because they package enyo-2 based versions as well.
```
How about opening an issue instead, and remove the whole block with the commented out code, including this TODO message? Personally I am not a fan of commented out code.

ollieparanoid commented on aports/luna/luna-firstuse/APKBUILD:
```diff
@@ -0,0 +1,36 @@
+package() {
+	cd "$srcdir"/build
+	make DESTDIR="$pkgdir" install
+
```
empty line

ollieparanoid commented on aports/luna/luna-init/APKBUILD:
```diff
@@ -28,6 +28,16 @@ build() {
 	make DESTDIR="$pkgdir" install
+
+	install -d "$pkgdir"/etc/palm
+    install -v -m 644 "$builddir"/files/conf/locale.txt "$pkgdir"/etc/palm
```
spaces

ollieparanoid commented on aports/luna/luna-systemui/APKBUILD:
```diff
@@ -0,0 +1,24 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webOS-ports/$pkgname/archive/$_commit.tar.gz"
+builddir="$srcdir/$pkgname-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/luna-temp-ofono-bluetooth/APKBUILD:
```diff
@@ -0,0 +1,24 @@
+source="luneos-components-$_commit.tar.gz::https://github.com/webOS-ports/luneos-components/archive/$_commit.tar.gz"
+builddir="$srcdir/luneos-components-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/mojoloader/APKBUILD:
```diff
@@ -0,0 +1,27 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webOS-ports/mojoloader/archive/$_commit.tar.gz"
+builddir="$srcdir/$pkgname-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/mojoloader/APKBUILD:
```diff
@@ -0,0 +1,27 @@
+pkgver=1.1_git20161005
+pkgrel=0
+_commit=4eec67e2c941ab416aaced46c604c2ff3747c2fd
+pkgdesc="mojoloader"
```
What does this program do?

ollieparanoid commented on aports/luna/mojoloader/APKBUILD:
```diff
@@ -0,0 +1,27 @@
+}
+
+package() {
+
```
empty lines

ollieparanoid commented on aports/luna/mojoservice-frameworks/APKBUILD:
```diff
@@ -0,0 +1,31 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webos-ports/mojoservice-frameworks/archive/$_commit.tar.gz"
+builddir="$srcdir/mojoservice-frameworks-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/mojoservice-frameworks/APKBUILD:
```diff
@@ -0,0 +1,31 @@
+
+	install -d "$pkgdir"/usr/palm/frameworks
+
+	for FRAMEWORK in `ls -d1 ./mojoservice*` ; do
```
`$()`?

ollieparanoid commented on aports/luna/nodejs-module-webos-pmlog/APKBUILD:
```diff
@@ -0,0 +1,44 @@
+license="GPL-3.0+"
+depends="pmloglib nodejs nodejs-npm"
+
+#vim is in makedepends because it provides xxd
```
missing space after "#"

ollieparanoid commented on aports/luna/webos-foundation/APKBUILD:
```diff
@@ -0,0 +1,31 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webos-ports/foundation-frameworks/archive/$_commit.tar.gz"
+builddir="$srcdir/foundation-frameworks-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/webos-loadable-frameworks/APKBUILD:
```diff
@@ -0,0 +1,31 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/webos-ports/loadable-frameworks/archive/$_commit.tar.gz"
+builddir="$srcdir/loadable-frameworks-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/luna/webos-underscore/APKBUILD:
```diff
@@ -0,0 +1,24 @@
+source="$pkgname-$_commit.tar.gz::https://github.com/openwebos/underscore/archive/$_commit.tar.gz"
+builddir="$srcdir/underscore-$_commit"
+
+build() {
```
empty build

ollieparanoid commented on aports/main/postmarketos-ui-luna/APKBUILD:
```diff
@@ -1,11 +1,11 @@
 arch="noarch"
 license="GPL3+"
-depends="luna-next-cardshell luna-next luna-sysmgr luna-sysservice luna-appmanager luneos-components luna-init mesa"
+depends="luna-next-cardshell luna-next luna-sysmgr luna-sysservice luna-appmanager luneos-components luna-init mesa db8 luna-webappmanager configurator activitymanager enyo luna-firstuse luna-systemui luna-qml-launcher luna-applauncher mojoloader luna-temp-ofono-bluetooth luna-app-services xkeyboard-config luna-core-apps luna-app-browser"
```
I recommend line breaks here. You could also put one package per line there, and sort them alphabetically.

ollieparanoid commented on aports/main/postmarketos-ui-luna/start_luna.sh:
```diff
@@ -1,27 +1,36 @@
+export QT_QPA_PLATFORM=wayland
+LunaAppManager -t -c -u luna &
+sleep 1
+LunaWebAppManager --verbose --allow-file-access-from-files
```
I know that this is just a workaround right now.
Do the daemons write anything to stdout or stderr? If so, I would recommend to redirect them to the syslog.

ollieparanoid replied:
@magmastonealex: thanks for being so fast!
This is the only thing left unanswered.

magmastonealex replied:
Ah, OK. I piped all of the ones with output to `logger` with tags, and verified they show up in `logread`.

ollieparanoid commented on aports/luna/luna-app-browser/APKBUILD:
```diff
@@ -0,0 +1,42 @@
+
+prepare() {
+	default_prepare
+	#cmake does not allow in-source builds
```
missing space after "#"

ollieparanoid commented on aports/luna/luna-firstuse/APKBUILD:
```diff
@@ -0,0 +1,36 @@
+
+prepare() {
+	default_prepare
+	#cmake does not allow in-source builds
```
missing space after "#"

ollieparanoid commented on aports/luna/luna-init/APKBUILD:
```diff
@@ -28,6 +28,16 @@ build() {
 
 package() {
 	cd "$srcdir"/build
+	#The cmake install file doesn't install everything in the right spot.
```
missing space after "#"

ollieparanoid commented on aports/luna/luna-webappmanager/APKBUILD:
```diff
@@ -0,0 +1,39 @@
+
+prepare() {
+	default_prepare
+	#cmake does not allow in-source builds
```
missing space after "#"

magmastonealex commented:
@ollieparanoid Unfortunately yes, `/usr/palm` is scattered all over the code, it'd be quite tricky to try and patch it out everywhere.

I think I've addressed everything else you brought up, and I removed all of the prepare statements that I added in this PR

montvid commented:
I just installed it from pmbootstrap menu. I can try again after this is merged.

montvid commented:
Oh @ollieparanoid disabled it for armhf?

magmastonealex commented:
@montvid, yes, disabled because qtwebengine doesn't build for armhf right now due to #1008. That means that Luna can't be used.

It makes sense you didn't have any success from the menu, you were missing everything that would have made it work :)
